### PR TITLE
Additional telemetry tracking

### PR DESCRIPTION
**Pull request by schustmi**
## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Describe changes
- Track integration installs and example pulls
- Refactor analytics events into enum
- Allow installation of multiple integrations at once, e.g. `zenml integration install airflow tensorflow`

### CODE REVIEW

**Review by htahir1 — COMMENTED** [review]
This LGTM. I really like Enum. Not sure about the changes you made here regarding requirements for integrations.

I was thinking of two more things to add here:

a) When users use Materializers
b) When users use Visualizers

It would help us get a sense of what materializers and visualizers people are liking? Is that possible or do you think too much

**Comment by schustmi** [comment]
I feel like that would be tracking on an individual class level which seems a little overboard to me, not sure if others would agree though. Should I create a ticket to discuss the general analytics strategy with some broader scope, i.e. to what granularity we would like to track usage of different parts of the codebase?

Regarding the integration install: I was just in the file and always thought it would have been nice to be able to install two integrations in a single call

File: tests/unit/cli/test_integration.py [hunk 1]
```diff
@@ -83,42 +89,78 @@ def test_integration_install_specific_integration(
...
+        sys.modules["zenml.cli.integration"],
+        "install_package",
+        return_value=None,
+    )
+    mocker.patch(
+        "zenml.integrations.registry.IntegrationRegistry.is_installed",
+        return_value=False,
+    )
+
+    result = runner.invoke(integration, ["install", "-f", *INTEGRATIONS])
```
Comment (AlexejPenner) [inline tests/unit/cli/test_integration.py:123]: Does this work on mark parametrize. As I understand it, mark.parametrize calls the test_function it decorates for each item in the supplied list.

Comment (AlexejPenner) [reply tests/unit/cli/test_integration.py:123]: As such I would have expected INTEGRATIONS to be a string here, so I would assume this iterates through the string?

**Review by AlexejPenner — COMMENTED** [review]
Looks great, also liking the enum for the analytics. Also like the easy implementation to allow installing multiple integrations. Just have one question regarding the tests, but would approve after :)

File: tests/unit/cli/test_integration.py (hunk 1 above)
Comment (AlexejPenner) [reply tests/unit/cli/test_integration.py:123]: My bad, I missread something, this function isn't decorated - ignore this

**Review by AlexejPenner — APPROVED** [review]

**Comment by schustmi** [comment]
@htahir1 Are you okay with merging this as is and creating a ticket for the general analytics strategy?

**Comment by htahir1** [comment]
Yes @schustmi lets go to develop!

**Pull request merged**